apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/environments/SidebarEnvironmentPanel.tsx

```diff
@@ -184,21 +198,27 @@ export const SidebarEnvironmentPanel: React.FC = () => {
               )
             </span>
             {node.data.targetFilter != null && (
-              <Link
-                href={`/${workspaceSlug}/targets?${new URLSearchParams({
-                  filter: LZString.compressToEncodedURIComponent(
-                    JSON.stringify(node.data.targetFilter),
-                  ),
-                })}`}
-                passHref
-              >
-                <Button variant="ghost" size="sm" asChild>
-                  <a target="_blank" rel="noopener noreferrer">
-                    <IconExternalLink className="mr-1 h-4 w-4" />
-                    View Targets
-                  </a>
-                </Button>
-              </Link>
+              <>
+                <TargetFilterUniquenessIndicator
+                  nodes={nodes}
+                  workspaceId={workspace.data?.id ?? ""}
+                />
+                <Link
+                  href={`/${workspaceSlug}/targets?${new URLSearchParams({
+                    filter: LZString.compressToEncodedURIComponent(
+                      JSON.stringify(node.data.targetFilter),
+                    ),
+                  })}`}
```

Comment:
this works you don't need to .toString it or something

Reply:
Yes, I think I had a .tostring there at one pint and you asked it to be removed? 

Either way it works.